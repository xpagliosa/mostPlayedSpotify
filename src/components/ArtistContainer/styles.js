import { StyleSheet } from 'react-native';
import { general, metrics, colors, fonts } from '../../styles';

const styles = StyleSheet.create({
    container: {
        minHeight: 145,
    }
})

export default styles;