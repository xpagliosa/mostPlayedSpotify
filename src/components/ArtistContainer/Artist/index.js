import React, { Component } from 'react';

import { Image, Text, View } from 'react-native';

import styles from './styles';

class Artist extends Component {

  render(){
    const { name, followers, images } = this.props;

    let foto = 'https://developer.spotify.com/wp-content/uploads/2016/07/icon3@2x.png';
    if (images.length > 0)
      foto = images[1].url

    return (
      <View style={styles.artist}>
        <Image
          style={styles.artistImage}
          source={{ uri: foto }}
        />
        <Text style={styles.artistName}>{name}</Text>
        <Text style={styles.artistFollow}>{followers.total} SEGUIDORES</Text>
      </View>
    )
  }
}

export default Artist;
