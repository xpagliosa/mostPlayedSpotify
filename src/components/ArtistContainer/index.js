import React, { Component } from 'react';

import { View } from 'react-native';

import api from '../../services/api';
import Artist from './Artist';
import Loading from '../Loading';

import styles from './styles';

class ArtistContainer extends Component {
    constructor(){
        super();
        this.state = {
            loading: true,
            loadingText: '',
            tryAgain: false,
            artist: [],
            artistID: '',
        }
    }

    async componentDidMount() {
        this.setLoading(true, "Buscando Artista...", false);
        await this.getArtist(this.props.artistID);
        this.setLoading(false, "", false);
    }

    async componentDidUpdate(prevProps, prevState) {
        if (prevProps.artistID !== this.props.artistID) {
            this.setLoading(true, "Buscando Artista...", false);
            await this.getArtist(this.props.artistID);
            this.setLoading(false, "", false);
        }
    }

    getArtist = async (artistID) => {
        try {
            const response = await api.get(`/artists/${artistID}`);
            // const axiosInstance = api.axiosInstance;
            const artist = response.data;
            await this.setState({ artist })
            this.setState({ artistID });
        } catch (err) {
            this.props.signIn();
        }
    }

    setLoading = (loading, loadingText, tryAgain) =>
    {
        this.setState({ loading });
        this.setState({ loadingText });
        this.setState({ tryAgain });
    }

    render(){
        const { loading, loadingText, tryAgain } = this.state;
        return (
            <View style={styles.container}>
                { loading 
                    ?   (
                            <Loading 
                                loadingText={loadingText}
                                tryAgain={tryAgain}
                                signIn={this.props.signIn}
                            />
                        )
                    :   ( 
                            <Artist 
                                {...this.state.artist}
                            />
                        )
                }
            </View>
        )
    }
}

export default ArtistContainer;