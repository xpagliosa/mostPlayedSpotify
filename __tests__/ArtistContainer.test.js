import 'react-native';
import React from 'react';
import { shallow } from 'enzyme';

import ArtistContainer from '../src/components/ArtistContainer';
import tagore from './tagore.json';
import confraria from './confraria.json';

describe('rendering ArtistContainer', () => {
    let wrapper = shallow(
        <ArtistContainer 
            loading={true}
            loadingText="Carregando Artista..."
            tryAgain={false}
            artistID="6jesZl7rJFJZ8d5i9pX5q8"
        />
    );
    it('shouls render Loading correctly', () => {
        wrapper.setState({ artist: [] });
        expect(wrapper).toMatchSnapshot();
    }),

    it('shouls render Tagore correctly', () => {
        wrapper.setState({ loading: false });
        wrapper.setState({ artist: tagore });
        expect(wrapper).toMatchSnapshot();
    }),

    it('shouls render Confraria correctly', () => {
        wrapper.setState({ loading: false });
        wrapper.setState({ artist: confraria });
        wrapper.setState({ artistID: "5slvIMrXIIfl1kNYszWP5G" });
        expect(wrapper).toMatchSnapshot();
    })
});

