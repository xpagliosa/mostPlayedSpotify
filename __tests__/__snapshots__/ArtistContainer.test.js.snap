// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering ArtistContainer shouls render Confraria correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ArtistContainer
    artist={
      Object {
        "external_urls": Object {
          "spotify": "https://open.spotify.com/artist/5slvIMrXIIfl1kNYszWP5G",
        },
        "followers": Object {
          "href": null,
          "total": 5773,
        },
        "genres": Array [],
        "href": "https://api.spotify.com/v1/artists/5slvIMrXIIfl1kNYszWP5G",
        "id": "5slvIMrXIIfl1kNYszWP5G",
        "images": Array [
          Object {
            "height": 640,
            "url": "https://i.scdn.co/image/bf38f72ca4fdd438381b25193d07d098cd568207",
            "width": 640,
          },
          Object {
            "height": 300,
            "url": "https://i.scdn.co/image/48d50a6b9914ec65d6f9cb7cea7646935bc79f38",
            "width": 300,
          },
          Object {
            "height": 64,
            "url": "https://i.scdn.co/image/8837012716d757c1867a121bef6dd6b375a41249",
            "width": 64,
          },
        ],
        "name": "Confraria da Costa",
        "popularity": 30,
        "type": "artist",
        "uri": "spotify:artist:5slvIMrXIIfl1kNYszWP5G",
      }
    }
    artistID="5slvIMrXIIfl1kNYszWP5G"
    loading={false}
    loadingText="Carregando Artista..."
    tryAgain={false}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <Loading
        loadingText="Buscando Artista..."
        signIn={undefined}
        tryAgain={false}
      />,
      "style": Object {
        "minHeight": 145,
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "loadingText": "Buscando Artista...",
        "signIn": undefined,
        "tryAgain": false,
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <Loading
          loadingText="Buscando Artista..."
          signIn={undefined}
          tryAgain={false}
        />,
        "style": Object {
          "minHeight": 145,
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "loadingText": "Buscando Artista...",
          "signIn": undefined,
          "tryAgain": false,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`rendering ArtistContainer shouls render Loading correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ArtistContainer
    artist={Array []}
    artistID="6jesZl7rJFJZ8d5i9pX5q8"
    loading={true}
    loadingText="Carregando Artista..."
    tryAgain={false}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <Loading
        loadingText="Buscando Artista..."
        signIn={undefined}
        tryAgain={false}
      />,
      "style": Object {
        "minHeight": 145,
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "loadingText": "Buscando Artista...",
        "signIn": undefined,
        "tryAgain": false,
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <Loading
          loadingText="Buscando Artista..."
          signIn={undefined}
          tryAgain={false}
        />,
        "style": Object {
          "minHeight": 145,
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "loadingText": "Buscando Artista...",
          "signIn": undefined,
          "tryAgain": false,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`rendering ArtistContainer shouls render Tagore correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ArtistContainer
    artist={
      Object {
        "external_urls": Object {
          "spotify": "https://open.spotify.com/artist/6jesZl7rJFJZ8d5i9pX5q8",
        },
        "followers": Object {
          "href": null,
          "total": 6212,
        },
        "genres": Array [],
        "href": "https://api.spotify.com/v1/artists/6jesZl7rJFJZ8d5i9pX5q8",
        "id": "6jesZl7rJFJZ8d5i9pX5q8",
        "images": Array [
          Object {
            "height": 640,
            "url": "https://i.scdn.co/image/b71f172b1d6c55024941d9ca72527b42ca581b31",
            "width": 640,
          },
          Object {
            "height": 300,
            "url": "https://i.scdn.co/image/89ea0966db59f494acf05caa057e423bcb0133c1",
            "width": 300,
          },
          Object {
            "height": 64,
            "url": "https://i.scdn.co/image/159ffd22713a6ab70363ea7198bed83fa240145f",
            "width": 64,
          },
        ],
        "name": "Tagore",
        "popularity": 32,
        "type": "artist",
        "uri": "spotify:artist:6jesZl7rJFJZ8d5i9pX5q8",
      }
    }
    artistID="6jesZl7rJFJZ8d5i9pX5q8"
    loading={false}
    loadingText="Carregando Artista..."
    tryAgain={false}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <Loading
        loadingText="Buscando Artista..."
        signIn={undefined}
        tryAgain={false}
      />,
      "style": Object {
        "minHeight": 145,
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "loadingText": "Buscando Artista...",
        "signIn": undefined,
        "tryAgain": false,
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <Loading
          loadingText="Buscando Artista..."
          signIn={undefined}
          tryAgain={false}
        />,
        "style": Object {
          "minHeight": 145,
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "loadingText": "Buscando Artista...",
          "signIn": undefined,
          "tryAgain": false,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
